CONTRIBUTING.md

```diff
@@ -111,3 +118,85 @@ and commits.
 ## Reporting Bugs and Creating Issues
 
 When opening a new issue, try to roughly follow the commit message format conventions above.
+
+
+# For developers
+
+## Building DBSP from sources
+
+DBSP is implemented in Rust and uses Rust's `cargo` build system. The SQL
+to DBSP compiler is implemented in Java and uses `maven` as its build system.
+
+You can build the rust sources by runnning the following at the top level of this tree.
+
+```
+cargo build
+```
+
+To build the SQL to DBSP compiler, run the following from `sql-to-dbsp-compiler/SQL-compiler`:
+
+```
+mvn package
+```
+
+If you want to develop DBSP without installing the required toolchains
+locally, you can use Github Codespaces; from
+https://github.com/feldera/dbsp, click on the green `<> Code` button,
+then select Codespaces and click on "Create codespace on main".
+
+## Learning the DBSP rust code
+
+To learn how the DBSP core works, we recommend starting with the tutorial.
+
+From the project root:
+
+```
+cargo doc --open
+```
+
+Then search for `dbsp::tutorial`.
+
+Another good place to start is the `circuit::circuit_builder` module documentation,
+or the examples folder.  For more sophisticated examples, try looking
+at the `nexmark` benchmark in the `benches` directory.
+
+## Running Benchmarks against DBSP
+
+The repository has a number of benchmarks available in the `benches` directory that provide a comparison of DBSP's performance against a known set of tests.
+
+Each benchmark has its own options and behavior, as outlined below.
+
+### Nexmark Benchmark
+
+You can run the complete set of Nexmark queries, with the default settings, with:
+
+```shell
+cargo bench --bench nexmark --features with-nexmark
+```
+
+By default this will run each query with a total of 100 million events emitted at 10M per second (by two event generator threads), using 2 CPU cores for processing the data.
+
+To run just the one query, q3, with only 10 million events, but using 8 CPU cores to process the data and 6 event generator threads, you can run:
+
+```shell
+cargo bench --bench nexmark --features with-nexmark -- --query q3 --max-events 10000000 --cpu-cores 8 --num-event-generators 6
+```
+
+For further options that you can use with the Nexmark benchmark,
+
+```shell
+cargo bench --bench nexmark --features with-nexmark -- --help
+```
+
+An extensive blog post about the implementation of Nexmark in DBSP:
+<https://liveandletlearn.net/post/vmware-take-3-experience-with-rust-and-dbsp/>
+
+
+## Updating the pipeline manager database schema
+
+Here are some guidelines when contributing code that affects the Pipeline Manager's DB schema.
```

Comment:
This is rather sudden. A couple of phrases explaining what the pipeline manager is?